core/vm/contracts.go

```diff
@@ -1135,3 +1142,35 @@ func kZGToVersionedHash(kzg kzg4844.Commitment) common.Hash {
 
 	return h
 }
+
+// P256VERIFY (secp256r1 signature verification)
+// implemented as a native contract
+type p256Verify struct{}
+
+// RequiredGas returns the gas required to execute the precompiled contract
+func (c *p256Verify) RequiredGas(input []byte) uint64 {
+	return params.P256VerifyGas
+}
+
+// Run executes the precompiled contract, returning the output and the used gas
+func (c *p256Verify) Run(input []byte) ([]byte, error) {
+	// Required input length is 160 bytes
+	const p256VerifyInputLength = 160
+
+	// "input" is (hash, r, s, x, y), each 32 bytes
+	input = common.RightPadBytes(input, p256VerifyInputLength)
```

Comment:
If the intended behavior is that the input must be exactly 160 bytes, then this padding here should not be applied.

Reply:
Thanks, applied!